vertical-pod-autoscaler/pkg/admission-controller/resource/vpa/handler.go

```diff
@@ -157,3 +170,19 @@ func validateVPA(vpa *vpa_types.VerticalPodAutoscaler, isCreate bool) error {
 
 	return nil
 }
+
+func validateResource(name corev1.ResourceName, val apires.Quantity) error {
+	switch name {
+	case corev1.ResourceCPU:
+		strValue := val.String()
```

Comment:
I think using [`AsScale`](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity.AsScale) will work better.
```
if _, precissionPreserved := val.AsScale(resource.Milli); !precissionPreserved {
 //...
```

For memory we're rounding to whole units, so there is no prefix ([docs](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Scale)), so you should use `AsScale(0)`.

Reply:
gotcha, I'm not familiar with this quantity object. thank you for your comments